client/components/data/sync-active-theme/index.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { waitFor } from 'calypso/my-sites/marketplace/util';
+import { requestActiveTheme } from 'calypso/state/themes/actions';
+import { isRequestingActiveTheme, getActiveTheme } from 'calypso/state/themes/selectors';
+
+interface SyncActiveThemeProps {
+	siteId: number;
+	themeId: string;
+	onThemeActive: CallableFunction;
+	maxAttempts?: number;
+}
+
+const SyncActiveTheme = ( {
+	siteId,
+	themeId,
+	onThemeActive,
+	maxAttempts = 20,
+}: SyncActiveThemeProps ) => {
+	const dispatch = useDispatch();
+	const requestStarted = useSelector( ( state ) => isRequestingActiveTheme( state, siteId ) );
+
+	const activeTheme = useSelector( ( state ) => getActiveTheme( state, siteId ) );
+
+	const [ attempts, setAttempts ] = useState( 0 );
+
+	useEffect( () => {
+		if ( activeTheme === themeId ) {
+			return onThemeActive();
+		}
+		if ( ! requestStarted && attempts < maxAttempts ) {
+			setAttempts( attempts + 1 );
+			waitFor( 5 ).then( () => dispatch( requestActiveTheme( siteId ) ) );
```

Comment:
Should the delay here be configurable? I feel like it may be better to check every 2-3 seconds to keep the delay as short as possible. We might need to bump `maxAttempts` up a bit to compensate, but I would rather us check more frequently so users don't have to wait some multiple of 5 seconds.

Reply:
Also, can we add a check for whether a transfer is in place, just in case somebody does land on the URL with the flag set to `true`, but the site isn't actually transferring?